processing/src/test/java/io/druid/query/groupby/GroupByQueryRunnerFactoryTest.java

```diff
@@ -86,6 +80,41 @@ public void testMergeRunnersEnsureGroupMerging() throws Exception
         )
         .build();
 
+    final QueryRunnerFactory factory = GroupByQueryRunnerTest.makeQueryRunnerFactory(new GroupByQueryConfig());
+
+    QueryRunner mergedRunner = factory.getToolchest().mergeResults(
+        new QueryRunner()
+        {
+          @Override
+          public Sequence run(Query query, Map responseContext)
+          {
+            return factory.getToolchest().mergeResults(
+                new QueryRunner()
+                {
+                  @Override
+                  public Sequence run(Query query, Map responseContext)
+                  {
+                    try {
+                      return new MergeSequence(
+                          query.getResultOrdering(),
+                          Sequences.simple(
+                              Arrays.asList(
+                                  factory.createRunner(createSegment()).run(query, responseContext),
+                                  factory.createRunner(createSegment()).run(query, responseContext)
+                              )
+                          )
+                      );
+                    } catch (Exception e) {
+                      Assert.fail(e.getMessage());
```

Comment:
Assert.fail inside a callback like this seems kinda weird -- usually asserts run inside the main test code. Would it work if this is just a re-throw? Like `throw Throwables.propagate(e)`